components/column-linking-table.vue

```diff
@@ -18,23 +19,75 @@
 
 <script>
 
+    // Allows for reference to store actions (index.js)
+    import { mapActions } from "vuex";
+
+    // Allows for reference to store data by creating simple, implicit getters
+    import { mapGetters } from "vuex";
+
     export default {
 
         props: {
 
-            categoryClasses: { type: Object, required: true },
-            columnToCategoryMap: { type: Object, required: true },
-            fields: { type: Array, required: true },
-            selectedCategory: { type: String, required: true },
-            tableData: { type: Array, required: true }
+            selectedCategory: { type: String, required: true }
+        },
+
+        data() {
+
+            return {
+
+                uiText: {
+
+                    tableFields: [
+
+                        { key: "column" },
+                        { key: "description" },
+                        { key: "category" }
+                    ]
+                }
+            };
+        },
+
+        computed: {
+
+            ...mapGetters([
+
+                "categoryClasses",
+                "columns",
+                "columnToCategoryMap"
+            ]),
+
+            tableRows() {
+
+                return this.columns.map(column => ({
+
+                    category: this.columnToCategoryMap[column],
+                    column: column.name,
+                    description: column.description
+                }));
+            }
         },
 
         methods: {
 
+            ...mapActions([
+
+                "alterColumnCategoryRelation"
+            ]),
+
             applyCategory(p_row, p_index, p_event) {
 
-                // Tell the parent page that a column has been linked with a category
-                this.$emit("column-name-selected", { column: p_row.column });
+                const payload = {
+
+                    category: this.selectedCategory,
+                    column: p_row.column
+                };
+
+                // 1. Link or unlink the currently-selected category and the clicked column
+                this.alterColumnCategoryRelation(payload);
+
+                // 2. Tell the categorization page a column has been clicked
+                this.$emit("column-name-clicked", payload);
```

Comment:
Did you still want to remove this emit here, now that the action will handle things?

Reply:
@surchs I was thinking of leaving the emit in place and removing it when I do the change over for next page accessibility with the layout/store (See my earlier [comment](https://github.com/neurobagel/annotation_tool/pull/230#discussion_r1029811986).

I suppose it doesn’t matter when the removal happens, but I thought removing it in this PR would be mixing issues.

Reply:
I was torn over it, and see the logic of both sides. Let's break things. I'll at least leave a `NOTE` in its place with a final commit and then merge.